### PR TITLE
Added stopping and traffic lights

### PR DESCRIPTION
**Pull request by matthewbartos**

### CODE REVIEW

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy [hunk 1]
```diff
@@ -11,8 +11,7 @@ import repast.simphony.relogo.schedule.Setup;
 import traffic.ReLogoPatch;
 
 class UserPatch extends ReLogoPatch{
-	static laneWidth = 4
-	
+
```
Comment (hejmsdz) [inline sz/Traffic/src/traffic/relogo/UserPatch.groovy:15]: This parameter is purely presentational (does not affect dynamics of the simulation), so I don't think it's beneficial to keep it as a configurable parameter. What do you think?

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy [hunk 2]
```diff
@@ -45,7 +44,28 @@ class UserPatch extends ReLogoPatch{
...
+		if (Math.abs(x) > laneWidth && Math.abs(y) > laneWidth) {
+			setPcolor(green())
+		} else if ((x == 0 || y == 0) && (x + y).intdiv(2) % 2 == 0) {
+			setPcolor(white())
+		} else {
+			setPcolor(gray() - 3)
+		}
+	}
+	
+	def changeLights(boolean areTrafficLightsGreenHorizontally ) {
```
Comment (hejmsdz) [inline sz/Traffic/src/traffic/relogo/UserPatch.groovy:49]: How about a shorter name?
```suggestion
	def changeLights(boolean greenHorizontally ) {
```

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy [hunk 3]
```diff
@@ -45,7 +44,28 @@ class UserPatch extends ReLogoPatch{
...
+		def x = getPxcor()
+		def y = getPycor()
+		
+		if (x == laneWidth + 1 && y == laneWidth) { // Right
+			areTrafficLightsGreenHorizontally ? setPcolor(lime()) : setPcolor(red())
+		} else if ((x == (laneWidth + 1) * -1) && (y == laneWidth * -1)) { // Left
```
Comment (hejmsdz) [inline sz/Traffic/src/traffic/relogo/UserPatch.groovy]: How about writing `-expression` instead of `expression * -1`?

**Review by hejmsdz — APPROVED** [review]

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy (hunk 2 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/UserPatch.groovy:49]: Yup, too much Java 😄

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy (hunk 1 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/UserPatch.groovy:15]: Sure!

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy (hunk 3 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/UserPatch.groovy]: Definitely :+1:

**Pull request merged**